src/ga/ga_selectors.rs

```diff
@@ -9,13 +9,11 @@ use std::cmp;
 /// Interface to Selection Schemes
 pub trait GASelector<'a, T: GASolution>
 {
-    /// Assign the population on which to operate
-    fn assign(&mut self, population: &'a mut GAPopulation<T>);
     /// Update internal state. 
     /// Some selectors implement an empty update().
-    fn update(&mut self) {}
+    fn update(&mut self, population : &mut GAPopulation<T>) {}
 
-    fn select(&self) -> &T;
+    fn select(&self, population: &'a GAPopulation<T>) -> &'a T;
 }
 
 pub trait GAScoreTypeBasedSelection<T: GASolution>
```

Comment:
@Sysnett I believe we can drop the generic type parameter from the trait and have it only at the function level, on functions that actually need it: score<T: GASolution>(), max_score<T: GASolution>(), and min_score<T: GASolution>(). The structs that implement the trait don't need to be parameterized, because they store no state that could be subject of parameterization.